application/controllers/QuestionAdministrationController.php

```diff
@@ -309,43 +309,100 @@ public function actionListQuestions($surveyid, $landOnSideMenuTab = 'settings')
         $aData['title_bar']['title']    = $oSurvey->currentLanguageSettings->surveyls_title .
             " (" . gT("ID") . ":" . $iSurveyID . ")";
 
-        $aData['topbar']['middleButtons'] = $this->renderPartial(
-            'partial/topbarBtns/listquestionsTopbarLeft_view',
-            [
-                'oSurvey' => $oSurvey,
-                'hasSurveyContentCreatePermission' => Permission::model()->hasSurveyPermission(
-                    $iSurveyID,
-                    'surveycontent',
-                    'create'
-                ),
-            ],
-            true
-        );
-
         // The DataProvider will be build from the Question model, search method
-        $model = new Question('search');
+        $questionModel = new Question('search');
         // Global filter
         if (isset($_GET['Question'])) {
-            $model->setAttributes($_GET['Question'], false);
+            $questionModel->setAttributes($_GET['Question'], false);
         }
         // Filter group
         if (isset($_GET['gid'])) {
-            $model->gid = $_GET['gid'];
+            $questionModel->gid = $_GET['gid'];
         }
         // Set number of page
         if (isset($_GET['pageSize'])) {
             App()->user->setState('pageSize', (int) $_GET['pageSize']);
         }
         $aData['pageSize'] = App()->user->getState('pageSize', App()->params['defaultPageSize']);
         // We filter the current survey id
-        $model->sid = $oSurvey->sid;
-        $aData['model'] = $model;
+        $questionModel->sid = $oSurvey->sid;
+        $aData['questionModel'] = $questionModel;
+
+        $aData['surveyid'] = $iSurveyID;
+        $aData['surveybar'] = [];
+        
+        // for newly combined groups and reorder parts
+        $aData['groupModel'] = $this->getGroupData($oSurvey);
+        $aData['aGroupsAndQuestions'] = $this->getReorderData($oSurvey);
 
         $this->aData = $aData;
 
+         $aData['hasSurveyContentCreatePermission'] = Permission::model()->hasSurveyPermission(
+            $iSurveyID,
+            'surveycontent',
+            'create'
+         );
+
+
         $this->render("listquestions", $aData);
     }
 
+    public function getGroupData($oSurvey) {
```

Comment:
Looks nice, but in PHP - by PSR-12 standard - we don't normally put `{` at the same line as the function name. This should be checked in CI tho, weird that I don't see it. It can also be adjusted automatically by the code-sniffer tool we use.